source/Mlos.Python/mlos/Optimizers/OptimizationProblem.py

```diff
@@ -99,20 +105,47 @@ def __init__(
 
         # Fit functions / surrogate models will be fed features consisting of both context and parameters.
         # Thus, the feature space is comprised of both context and parameters.
+        has_context = self.context_space is not None
         self.feature_space = SimpleHypergrid(
             name="features",
             dimensions=[
-                CategoricalDimension(name="contains_parameters", values=[True]),
-                CategoricalDimension(name="contains_context", values=[self.context_space is not None])
+                CategoricalDimension(name="contains_context", values=[has_context])

```

Comment:
I changed it to having only one join dimension, assuming we always have parameters.

Reply:
Makes sense.  `assert` that to help the user?

Reply:
Also do you need to remove `contains_parameters` from the `META_DIMENSIONS_NAMES` above?